virtinst/guest.py

```diff
@@ -785,6 +785,13 @@ def sync_vcpus_topology(self, defCPUs):
                 self.vcpus = defCPUs
         self.cpu.set_topology_defaults(self.vcpus)
 
+    def change_graphics(self, val, inst):
+        inst.type = val
+        if val == "spice":
+            self._add_spice_devices()
+        else:
+            self._remove_spice_devices()
```

Comment:
coverage points out we don't hit this line. so we need a cli test case for switching spice -> vnc. Preferrably as a prep commit, so the diff here also demonstrates the XML bits changed. maybe test-for-virtxml isn't sufficient for input and we need a designated domain for it

Reply:
Yeah we should test it. Will do and include video change as well into that test case so it demonstrate the real world usage.